master/internal/rm/kubernetesrm/kubernetes_resource_manager.go

```diff
@@ -193,7 +193,9 @@ func (k *ResourceManager) GetAgent(msg *apiv1.GetAgentRequest) (*apiv1.GetAgentR
 
 // GetAgents implements rm.ResourceManager.
 func (k *ResourceManager) GetAgents() (*apiv1.GetAgentsResponse, error) {
-	return k.jobsService.GetAgents(), nil
+	resp := k.jobsService.GetAgents()
+	// Ensure cached response is not inadvertently modified.
+	return rm.CopyGetAgentsResponse(resp)
```

Comment:
probably is safer to do this copy on ```jobsService.getAgents``` in case someone down the line calls this within the jobs service

Reply:
By that do you mean, copy the response before returning from `jobsService.GetAgents()` ?

I had that originally, but the function signature didn't return an error. Thinking about it, there's no reason I can't update the jobsService function to return an error. :)